### PR TITLE
OCL_ICD: new recipe

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: dev-libs/ocl-icd/ocl_icd-2.2.12.recipe [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+DESCRIPTION="OpenCL implementations are provided as ICD (Installable Client \
+Driver). An OpenCL program can use several ICD thanks to the use of an ICD \
+Loader as provided by this project. This free ICD Loader can load any (free \
+or non free) ICD."
+HOMEPAGE="http://www.blosc.org/"
+COPYRIGHT="2012-2015 Brice Videau
+	2012-2015 Vincent Danjean"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://forge.imag.fr/frs/download.php/836/ocl-icd-2.2.12.tar.gz"
```
Comment (fbrosson) [inline dev-libs/ocl-icd/ocl_icd-2.2.12.recipe]:
```diff
-SOURCE_URI="https://forge.imag.fr/frs/download.php/836/ocl-icd-2.2.12.tar.gz"
+SOURCE_URI="https://forge.imag.fr/frs/download.php/836/ocl-icd-$portVersion.tar.gz"
```
Hmm, there's the "836" which will probably change, so maybe you should ignore this suggestion.

File: dev-libs/ocl-icd/ocl_icd-2.2.12.recipe [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	cmd:awk
+	cmd:diff
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:ruby
+	"
+
+defineDebugInfoPackage ocl_icd$secondaryArchSuffix \
+	$libDir/libOpenCL.so.$libVersion
```
Comment (fbrosson) [inline dev-libs/ocl-icd/ocl_icd-2.2.12.recipe]: Suggestion:
```diff
-	$libDir/libOpenCL.so.$libVersion
+	'$libDir"/libOpenCL.so.$libVersion
```

File: dev-libs/ocl-icd/ocl_icd-2.2.12.recipe [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	make install
+
+	# prepare develop/lib
+	prepareInstalledDevelLibs libOpenCL
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel \
+		$developDir \
+		$docDir
```
Comment (fbrosson) [inline dev-libs/ocl-icd/ocl_icd-2.2.12.recipe]: Suggestion:
```diff

 	packageEntries devel \
-		$developDir \
+		"$developDir" \
-		$docDir
+		"$docDir"
```

File: dev-libs/ocl-icd/ocl_icd-2.2.12.recipe [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+HOMEPAGE="http://www.blosc.org/"
+COPYRIGHT="2012-2015 Brice Videau
+	2012-2015 Vincent Danjean"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://forge.imag.fr/frs/download.php/836/ocl-icd-$portVersion.tar.gz"
+CHECKSUM_SHA256="7665f368354e3d2b7787ba4a23c6f061db1181195ba1914dd1cdcd462eca4df4"
+SOURCE_DIR="ocl-icd-$portVersion"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
```
Comment (fbrosson) [inline dev-libs/ocl-icd/ocl_icd-2.2.12.recipe]: Maybe:
```diff
-ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
```
It looks like we would need a rather big patch for `ocl_icd_loader.c` to compile with gcc2, so maybe we can mark it as broken on x86_gcc2.
That said, I'm OK with keeping the `?x86_gcc2`.

File: dev-libs/ocl-icd/ocl_icd-2.2.12.recipe [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:diff
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:ruby
+	"
+
+defineDebugInfoPackage ocl_icd$secondaryArchSuffix \
```
Comment (fbrosson) [inline dev-libs/ocl-icd/ocl_icd-2.2.12.recipe]:
```diff
 	cmd:ruby
 	"
 
+TEST_REQUIRES="
+ 	cmd:find
+ 	"
+
 defineDebugInfoPackage ocl_icd$secondaryArchSuffix \
```

This allows to get rid of some warnings: `./testsuite: line 1660: find: command not found`
<Details>

```
## ------------------------------------ ##
## OpenCL ICD Loader 2.2.12 test suite. ##
## ------------------------------------ ##

Standard compliant ocl-icd libOpenCL.so

  1: invalid OCL_ICD_VENDORS                         ok
./testsuite: line 1660: find: command not found
  2: OCL_ICD_VENDORS as directory                    ok
./testsuite: line 1660: find: command not found
  3: OCL_ICD_VENDORS as library file                 ok
./testsuite: line 1660: find: command not found
```
...

</Details>

File: dev-libs/ocl-icd/ocl_icd-2.2.12.recipe (hunk 4 above)
Comment (extrowerk) [reply dev-libs/ocl-icd/ocl_icd-2.2.12.recipe]: It is absolutely not needed on gcc2. But thanks for the test.

**Comment by extrowerk** [comment]
@alviss: nope, it is example

**Comment by alaviss** [comment]
> @alviss

It's @alaviss :wink: 

> nope, it is example

Then please put them into `developDocDir`, since they're development-related documentations

**Comment by extrowerk** [comment]
@alaviss: aye aye.

**Pull request merged**